spectrum-network/src/peer/peer_state.rs

```diff
@@ -0,0 +1,89 @@
+use std::borrow::{Borrow, Cow};
+use libp2p::PeerId;
+use crate::peer::data::{PeerConnState, PeerInfo};
+use crate::peer::peer_store::{PeerStore, PeerStoreRejection};
+use crate::peer::types::Reputation;
+
+pub struct ConnectedPeer<'a> {
+    peer_id: Cow<'a, PeerId>,
```

Comment:
`PeerId` is a copy type. That fact can bring lots of simplicity to the code by removing the lifetimes and `Cow`.